src-java/server42/server42-control-storm-topology/src/main/java/org/openkilda/server42/control/topology/storm/bolt/flow/FlowHandler.java

```diff
@@ -0,0 +1,142 @@
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.server42.control.topology.storm.bolt.flow;
+
+import org.openkilda.messaging.payload.flow.FlowEndpointPayload;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.server42.control.messaging.flowrtt.AddFlow;
+import org.openkilda.server42.control.messaging.flowrtt.ClearFlows;
+import org.openkilda.server42.control.messaging.flowrtt.Headers;
+import org.openkilda.server42.control.messaging.flowrtt.RemoveFlow;
+import org.openkilda.server42.control.topology.service.FlowRttService;
+import org.openkilda.server42.control.topology.service.IFlowCarrier;
+import org.openkilda.server42.control.topology.storm.ComponentId;
+import org.openkilda.server42.control.topology.storm.bolt.flow.command.FlowCommand;
+import org.openkilda.server42.control.topology.storm.bolt.router.Router;
+import org.openkilda.server42.messaging.FlowDirection;
+import org.openkilda.wfm.AbstractBolt;
+import org.openkilda.wfm.error.PipelineException;
+
+import org.apache.storm.kafka.bolt.mapper.FieldNameBasedTupleToKafkaMapper;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.apache.storm.tuple.Values;
+
+public class FlowHandler extends AbstractBolt
+        implements IFlowCarrier {
+
+    public static final String BOLT_ID = ComponentId.FLOW_HANDLER.toString();
+
+    public static final String STREAM_CONTROL_COMMANDS_ID = "control.commands";
+    public static final Fields STREAM_CONTROL_COMMANDS_FIELDS = new Fields(FieldNameBasedTupleToKafkaMapper.BOLT_KEY,
+            FieldNameBasedTupleToKafkaMapper.BOLT_MESSAGE);
+    private final PersistenceManager persistenceManager;
+    private transient FlowRttService flowRttService;
+
+    public FlowHandler(PersistenceManager persistenceManager) {
+        this.persistenceManager = persistenceManager;
+    }
+
+    protected void init() {
+        this.flowRttService = new FlowRttService(this, persistenceManager);
+    }
+
+    @Override
+    protected void handleInput(Tuple input) throws Exception {
+        String source = input.getSourceComponent();
+        if (Router.BOLT_ID.equals(source)) {
+            handleRouterCommand(input);
+        } else {
+            unhandledInput(input);
+        }
+    }
+
+    @Override
+    protected void handleException(Exception error) throws Exception {
```

Comment:
Abstract bolt already does this (with the only difference - it will add "Unhandled exception prefix" to the log message).

Reply:
done